USF-1440: Checkout feature matrix

jeff-matthews opened the pull request:
This pull request (PR) adds a table showing which Adobe Commerce features that the checkout dropin supports.

It also hides incomplete checkout content. We should avoid publishing blank pages in the future.

>[!NOTE]
>The `OptionsTable` component doesn't currently support image links. It would be nice to have icons/.svg files for each status similar to [this](https://developer.adobe.com/commerce/pwa-studio/integrations/adobe-commerce/features/).

I know that we're attempting to finalize the dropin doc template (#41), but since we'll have to refactor the cart docs after that anyway, it seems prudent to get this info out on production. It's unclear how long it will take to get the template finalized.

jeff-matthews committed:
docs: added initial draft of checkout feature matrix

jeff-matthews committed:
chore: hide incomplete checkout topics

bdenham commented on src/content/docs/dropins/checkout/index.mdx:
```diff
@@ -7,14 +7,46 @@ sidebar:
-<Aside type="wip" title="In Development">
-  The Checkout dropin is in active development. Stay tuned for updates.
-</Aside>
+The following table provides an overview of the Adobe Commerce features that the Checkout dropin supports:
```
```suggestion
## Commerce feature support

The following table provides an overview of the Adobe Commerce features that the Checkout dropin supports:
```

bdenham reviewed (approved):
I suggested a heading for the table. Otherwise, looks great.

<img width="845" alt="image" src="https://github.com/user-attachments/assets/c2e9f0ee-7e8f-43f3-be89-429a47d90e59">

jeff-matthews committed:
docs: apply suggestions from code review

Co-authored-by: Bruce Denham <bdenham@adobe.com>

jeff-matthews committed:
fix: alphabetized list of features

keharper commented:
How about "Supported Commerce features" instead of "Commerce features supported"? That feels more natural to me, but no biggie either way

jeff-matthews committed:
docs: apply suggestions from code review

jeff-matthews committed:
Merge branch 'dropin-feature-matrices' into USF-1440-checkout-feature-matrix

jeff-matthews committed:
fix: converted to markdown tables w/ badges

jeff-matthews commented:
I converted to the default Starlight table component and added badges. It required some minor style adjustments, which I [made](https://github.com/commerce-docs/microsite-commerce-storefront/commit/f121f7f02a121378a2ffe8f7397e974f505207ab) in a separate integration branch (now the base branch for this PR):

![checkout](https://github.com/user-attachments/assets/7c017756-ab31-4c83-81ca-807bde43aa3a)

bdenham reviewed (approved):
I like the use of the Badge components here! They give the statuses a quick visual representation while providing the text for meaning so that the user does not have to guess. Good thinking!

jeff-matthews commented:
Approved by Olena in Slack
